src/ukify/ukify.py

```diff
@@ -858,6 +893,57 @@ def generate_keys(opts):
             pub_key.write_bytes(pub_key_pem)
 
 
+def pe_read_single_section(section, opts, name, text):
+    data = section.get_data(ignore_padding=True)
+    size = section.Misc_VirtualSize
+    h = sha256(data).hexdigest()
+    data_printed = None
+
+    if text:
+        try:
+            data_printed = data.decode()
+        except UnicodeDecodeError:
+            print(f"Warning! {name} cannot be read")
+            data_printed = None
+
+    tmp = {
+        'size' : size,
+        'sha256' : h,
+    }
+
+    if text:
+        tmp['text'] = data_printed
```

Comment:
Why not merge this `if` with the previous one?

If we expect a section to be text, but it is not valid unicode, the output should say so explicitly. So e.g. "(not valid UTF-8-encoded text)" or something like that.